web/e2e-tests/stream_create.test.ts

```diff
@@ -74,15 +74,27 @@ async function test_user_filter_ui(page: Page): Promise<void> {
     await await_user_visible(page, "othello");
 }
 
-async function create_stream(page: Page): Promise<void> {
+async function create_stream(page: Page, first_stream: boolean): Promise<void> {
     await page.waitForSelector('xpath///*[text()="Create stream"]', {visible: true});
-    await common.fill_form(page, "form#stream_creation_form", {
-        stream_name: "Puppeteer",
-        stream_description: "Everything Puppeteer",
-    });
+    const test_stream = {
+        stream_name: "Puppeteer" + (first_stream ? "" : "2"),
+        stream_description: "Everything Puppeteer" + (first_stream ? "" : "2"),
+    };
+    await common.fill_form(page, "form#stream_creation_form", test_stream);
     await page.click("form#stream_creation_form .finalize_create_stream");
+    if (first_stream) {
+        // an explanatory modal is shown for the first stream created
+        await page.waitForSelector(".modal__btn.dialog_exit_button.animated-purple-button");
+        await page.click(".modal__btn.dialog_exit_button.animated-purple-button");
+    }
+    await page.waitForSelector(".message-header-stream-settings-button.tippy-zulip-tooltip", {
+        visible: true,
+    });
+    await page.click(".message-header-stream-settings-button.tippy-zulip-tooltip");
+    await new Promise((r) => setTimeout(r, 1000));
```

Comment:
I'm not sure why but not waiting for a while resulted in the failure of the test.

Reply:
One never wants to do sleeps in an e2e test like this -- figure out what condition to wait for and waiting for the appropriate selector instead. 

(Otherwise it's fail nondeterministically).

Reply:
I've figured that out and removed the sleep. The tests now pass deterministically.